src/Processors/Sources/MongoDBSource.cpp

```diff
@@ -381,8 +379,9 @@ MongoDBCursor::MongoDBCursor(
     const std::string & collection,
     const Block & sample_block_to_select,
     const Poco::MongoDB::Document & query,
-    Poco::MongoDB::Connection & connection)
-    : is_wire_protocol_old(isMongoDBWireProtocolOld(connection))
+    Poco::MongoDB::Connection & connection,
+    bool projections_)
```

Comment:
Maybe we can just check if `sample_block_to_select` is empty instead of adding new flag here?